shell/components/ClusterIconMenu.vue

```diff
@@ -12,6 +12,14 @@ export default {
     isEnabled() {
       return !!this.cluster?.ready;
     },
+    showLocalIcon() {
+      return this.cluster.isLocal && !this.cluster.isHarvester && !this.cluster.badge?.iconText;
+    },
+    badgeLogoBorderBottom() {
+      const { color } = this.cluster.badge;
```

Comment:
It seems like badge is not always present in the pre-existing code (and we don't have types), so to destructure it will cause error.

Reply:
good catch, thanks!